pandas/tests/groupby/test_value_counts.py

```diff
@@ -54,6 +54,8 @@ def seed_df(seed_nans, n, m):
             "3rd": np.random.randint(1, m + 1, n),
         }
     )
+    # Explicitly cast to float to avoid implicit cast when setting nan
+    frame["3rd"] = frame["3rd"].astype("float")
```

Comment:
Shouldn't this be needed only if this hits `if seed_nans`?

Reply:
you're right, thanks!